src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+    }
+
+    //-----------------------------------------------------------------------
+    /**
+     * Gets the chronology of this date, which is the International fixed calendar system.
+     * <p/>
+     * The {@code Chronology} represents the calendar system in use.
+     * The era and other fields in {@link ChronoField} are defined by the chronology.
+     *
+     * @return the International fixed chronology, not null
+     */
+    @Override
+    public InternationalFixedChronology getChronology() {
+        return InternationalFixedChronology.INSTANCE;
+    }
+
+    /**
+     * Gets the era applicable at this date.
+     * <p/>
+     * The International fixed calendar system only has one era, 'CE',
+     * defined by {@link InternationalFixedEra}.
+     *
+     * @return the era applicable at this date, not null
+     */
+    @Override
+    public InternationalFixedEra getEra() {
+        return InternationalFixedEra.CE;
+    }
+
+    /**
+     * Returns the length of the month represented by this date.
+     * <p/>
+     * This returns the length of the month in days.
+     * Month lengths do not match those of the ISO calendar system.
+     *
+     * @return the length of the month in days: 28
+     */
+    @Override
+    public int lengthOfMonth() {
+        return InternationalFixedChronology.DAYS_IN_MONTH;
+    }
+
+    /**
+     * Returns the length of the year represented by this date.
+     * <p/>
+     * This returns the length of the year in days.
+     * Year lengths match those of the ISO calendar system.
+     *
+     * @return the length of the year in days: 365 or 366
+     */
+    @Override
+    public int lengthOfYear() {
+        return InternationalFixedChronology.DAYS_IN_YEAR + (isLeapYear() ? 1 : 0);
+    }
+
+    @Override
+    public InternationalFixedDate plus(final long amountToAdd, final TemporalUnit unit) {
+        if (amountToAdd == 0) {
+            return this;
+        }
+
+        if (unit instanceof ChronoUnit) {
+            ChronoUnit f = (ChronoUnit) unit;
+
+            switch (f) {
+                case DAYS:
+                    return plusDays(amountToAdd);
+                case MONTHS:
+                    return plusMonths(amountToAdd);
+                case YEARS:
+                    return plusYears(amountToAdd);
+                default:
+                    break;
+            }
+        }
+
+        return (InternationalFixedDate) super.plus(amountToAdd, unit);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    InternationalFixedDate plusYears(final long yearsToAdd) {
+        if (yearsToAdd == 0) {
+            return this;
+        }
+
+        int newYear = (int) Math.addExact(getProlepticYear(), yearsToAdd);
+
+        if (isYearDay()) {
+            return create(newYear, false, true);
+        }
+
+        int day = getDayOfMonth();
+        int newMonth = getCalculatedMonth();
+
+        if (isLeapDay()) {
+            if ((yearsToAdd & 3) == 0) {
```

Comment:
Touché on the century leap-year bug, corrected it now.

However, Abstract#plusYears(long) performs a check against ValueRange.of(Year.MIN_VALUE, Year.MAX_VALUE), which is not useful for InternationalFixedDate.
There are appropriate year checks in the factory methods though.

Got rid of plusYears(long) and adapted resolvePreviousValid(int,int,int).